flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -1512,18 +1513,52 @@ public void createTemporaryModel(
     /**
      * Alters a model in a given fully qualified path.
      *
-     * @param modelChange The model containing only changes
+     * @param newModel The new model containing changes. It could be null if the model to alter
+     *     doesn't exist
+     * @param modelChanges The changes to apply to the model.
      * @param objectIdentifier The fully qualified path where to alter the model.
      * @param ignoreIfNotExists If false exception will be thrown if the model to be altered does
      *     not exist.
      */
     public void alterModel(
-            CatalogModel modelChange,
+            @Nullable CatalogModel newModel,
+            List<ModelChange> modelChanges,
             ObjectIdentifier objectIdentifier,
             boolean ignoreIfNotExists) {
         execute(
                 (catalog, path) -> {
-                    ResolvedCatalogModel resolvedModel = resolveCatalogModel(modelChange);
+                    ResolvedCatalogModel resolvedModel =
+                            newModel == null ? null : resolveCatalogModel(newModel);
+                    catalog.alterModel(path, resolvedModel, modelChanges, ignoreIfNotExists);
```

Comment:
I wonder if there is a reason to pass non-existing model as a `null` further rather than applying similar behavior as for `alterTable`?

Reply:
In `altertable` call, if the table doesn't exist, the operation is a `NopOperation`: https://github.com/apache/flink/blob/master/flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/SqlNodeToOperationConversion.java#L409. This leads to problem where we don't know the original sql operation and properly give error message etc. So in `alterModel` it's still `altermodelOperation` and null indicates original model doesn't exist. It can be handled by downstream

Reply:
We cannot simply pass nulls if this is not documented in the Catalog JavaDoc. Why should we allow alter model if it doesn't exist? The catalog manager or higher layers can deal with the problem.

Reply:
made it unresolved again since the problem is not solved yet